vip-jetpack/connection-pilot/class-jetpack-connection-pilot.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+/**
+ * The Pilot is in control of setting up the cron job for monitoring JP connections and sending out alerts if anything is wrong.
+ * Will only run if the `WPCOM_VIP_RUN_CONNECTION_PILOT` constant is defined and set to true.
+ */
+class WPCOM_VIP_Jetpack_Connection_Pilot {
+
+	/**
+	 * Cron action that runs the connection pilot checks.
+	 */
+	const CRON_ACTION = 'wpcom_vip_run_jetpack_connection_pilot';
+
+	/**
+	 * The schedule the cron job runs on. Update in 000-vip-init.php as well.
+	 *
+	 * Schedule changes can take up to 24 hours to take effect.
+	 * See the a8c_cron_control_clean_legacy_data event for more details.
+	 */
+	const CRON_SCHEDULE = 'hourly';
+
+	/**
+	 * Initiate an instance of this class if one doesn't exist already.
+	 */
+	public static function init() {
+		static $instance = false;
+
+		if ( ! $instance ) {
+			$instance = new WPCOM_VIP_Jetpack_Connection_Pilot();
+		}
+
+		return $instance;
+	}
+
+	/**
+	 * Class constructor.
+	 * Ensures the cron job is set up correctly and ready.
+	 */
+	private function __construct() {
+		if ( ! self::should_run_connection_pilot() ) {
+			return;
+		}
+
+		// Ensure the internal cron job has been added. Should already exist as an internal Cron Control job.
+		if ( ! has_action( self::CRON_ACTION ) ) {
```

Comment:
Took inspiration from the other internal cron-control event on this part: https://github.com/Automattic/vip-support/blob/00dcbc16b1a9fc54a4fe2dc9075edf6f570d7ea7/class-vip-support-user.php#L166

Cron-control should be declaring the action, so it seemed like a good check to ensure cron-control is working as needed.

Reply:
But what happens if, in 6 months, we decide that we want something new to run on `wpcom_vip_run_jetpack_connection_pilot`, so we add:

```
add_action( 'wpcom_vip_run_jetpack_connection_pilot', 'do_stuff' )
```

in some other file? We'll unknowingly break the main pilot code because now there's already an action registered, so `has_action()` returns `true`, and we never register this action.

Not saying that's a likely scenario, but it makes this code fragile if simply hooking into the same action anywhere else will cause it to stop working.

Reply:
100b1ecc